source/docs/appendices/migration-guides.md

```diff
@@ -7,6 +7,162 @@ This area will contain the Migration steps to follow for upgrading your store to
 
 Our goal is to make migrations as smooth as possible. This is why we try to make many changes backward compatible by using deprecation warnings. The deprecation warnings are usually removed in the next breaking release.
 
+## `2.3.x` -> `2.4.0`
+
+<blockquote class="important">
+  **We recommend that you first follow the [`2.2.x` -> `2.3.0` migration guide](#2-2-x-gt-2-3-0).**
+  This release was mostly focused on a new optional base theme, and there are not many upgrade instructions.
+</blockquote>
+
+### Cart item may not have product data attached
+
+We fixed some issues in case products added to cart went out of stock. Our default components were not resilient to Cart items not having an attached product anymore and led to errors.
```

Comment:
The twe sentences deal about two different issues. Out of stock does not mean the product disappears. The disappearance issue was related to multiple website issues.

In the current form of this paragraph, it seems like the two issues are linked.